cmd/notary-server/config.go

```diff
@@ -61,25 +63,44 @@ func grpcTLS(configuration *viper.Viper) (*tls.Config, error) {
 }
 
 // parses the configuration and returns a backing store for the TUF files
-func getStore(configuration *viper.Viper, allowedBackends []string, hRegister healthRegister) (
+func getStore(configuration *viper.Viper, hRegister healthRegister) (
 	storage.MetaStore, error) {
+	var (
+		store storage.MetaStore
+		err   error
+	)
+	backend := configuration.GetString("storage.backend")
+	logrus.Infof("Using %s backend", backend)
 
-	storeConfig, err := utils.ParseStorage(configuration, allowedBackends)
-	if err != nil {
-		return nil, err
-	}
-	logrus.Infof("Using %s backend", storeConfig.Backend)
-
-	if storeConfig.Backend == utils.MemoryBackend {
+	switch backend {
+	case notary.MemoryBackend:
 		return storage.NewMemStorage(), nil
+	case notary.MySQLBackend, notary.SQLiteBackend:
+		storeConfig, err := utils.ParseSQLStorage(configuration)
+		if err != nil {
+			return nil, err
+		}
+		s, err := storage.NewSQLStorage(storeConfig.Backend, storeConfig.Source)
+		store = *storage.NewTUFMetaStorage(s)
+		hRegister("DB operational", s.CheckHealth, time.Second*60)
+	case notary.RethinkDBBackend:
+		var sess *gorethink.Session
+		storeConfig, err := utils.ParseRethinkDBStorage(configuration)
+		if err != nil {
+			return nil, err
+		}
+		sess, err = rethinkdb.Connection(storeConfig.CA, storeConfig.Source)
```

Comment:
Can we add a negative test for what happens if we can't connect (tls cert doesn't exist, or can't connect to server)?